zio-http/src/main/scala/zhttp/http/Cookie.scala

```diff
@@ -147,25 +147,46 @@ object Cookie {
     val (first, other)    = (cookieWithoutMeta.head, cookieWithoutMeta.tail)
     val (name, content)   = splitNameContent(first)
 
-    val cookie =
-      if (name.trim == "" && content.isEmpty) Option.empty[Cookie]
-      else Some(Cookie(name, content.getOrElse("")))
-
-    other.map(splitNameContent).map(t => (t._1.toLowerCase, t._2)).foldLeft(cookie) {
-      case (Some(c), ("expires", Some(v)))  => parseDate(v).map(c.withExpiry(_))
-      case (Some(c), ("max-age", Some(v)))  => Try(v.toLong).toOption.map(c.withMaxAge(_))
-      case (Some(c), ("domain", v))         => Some(c.withDomain(v.getOrElse("")))
-      case (Some(c), ("path", v))           => Some(c.withPath(Path(v.getOrElse(""))))
-      case (Some(c), ("secure", _))         => Some(c.withSecure)
-      case (Some(c), ("httponly", _))       => Some(c.withHttpOnly)
-      case (Some(c), ("samesite", Some(v))) =>
-        v.trim.toLowerCase match {
-          case "lax"    => Some(c.withSameSite(SameSite.Lax))
-          case "strict" => Some(c.withSameSite(SameSite.Strict))
-          case "none"   => Some(c.withSameSite(SameSite.None))
-          case _        => Some(c)
-        }
-      case (c, _)                           => c
+    if (name.trim != "" || content.isDefined) {
+      var expires: Option[Instant]          = None
+      var maxAge: Option[Long]              = None
+      var domain: Option[String]            = None
+      var path: Option[Path]                = None
+      var secure: Boolean                   = false
+      var httpOnly: Boolean                 = false
+      var sameSite: Option[Cookie.SameSite] = None
+
+      other.map(splitNameContent).map(n => (n._1.toLowerCase, n._2)).foreach {
+        case ("expires", Some(v))  => expires = parseDate(v)
+        case ("max-age", Some(v))  => maxAge = Try(v.toLong).toOption
+        case ("domain", v)         => domain = v
+        case ("path", v)           => path = v.map(Path(_))
+        case ("secure", _)         => secure = true
+        case ("httponly", _)       => httpOnly = true
+        case ("samesite", Some(v)) =>
+          v.trim.toLowerCase match {
+            case "lax"    => sameSite = Some(SameSite.Lax)
+            case "strict" => sameSite = Some(SameSite.Strict)
+            case "none"   => sameSite = Some(SameSite.None)
+            case _        => ()
+          }
+        case (_, _)                => ()
+      }
+      Some(
+        Cookie(
+          name = name,
+          content = content.getOrElse(""),
+          expires = expires,
+          maxAge = maxAge,
+          domain = domain,
+          path = path,
+          isSecure = secure,
+          isHttpOnly = httpOnly,
+          sameSite = sameSite,
+        ),
+      )
+    } else {
+      None
```

Comment:
This is great, you have almost doubled the perf! 

You can further reduce allocations by — 
1. Get rid of all the `Option` and use `null` instead. 
2. Every `map` operation is creating a new collection (more memory), use a for/while loop if possible
3. Get rid of interim `Function` and `Tuple` also.

Reply:
Made some changes to avoid string split and some allocations.

Marked internal functions _parseDate_ and _splitNameContent_ @inline functions, not sure if it could be a problem ...

Reply:
Instead of using `None` use `null`. This can cause null pointer exceptions but that should be fine because we will rename this method to `unsafeDecodeResponseCookie` 
and make it package-private. And, we will have a separate method `decodeResponseCookie` which calls `unsafeDecodeResponseCookie` and wraps the output in an `Option`. And that  should be a public method.